### PR TITLE
[W05H03] Added Tests for w05h03 Messenger

### PR DESCRIPTION
**Pull request by MaximilianBoehmichen**
nach meinem Verständnis testet das einige Edge-cases / Sonderfälle und verrät keine Lösung

### CODE REVIEW

**Comment by cod-eric** [comment]
Looks good, optional tests might throw off some people, but `[OPTIONAL]` notation in their display name should be fine.

**Review by cod-eric — APPROVED** [review]
👍

**Review by LeoWoehler — APPROVED** [review]
die Tests laufen alle bei mir und haben meiner meinung nach überall die richtigen ergebnisse

File: w05h03/MessengerTests.java [hunk 1]
```diff
@@ -0,0 +1,319 @@
+package Test;
```
Comment (nilsreichardt) [inline w05h03/MessengerTests.java:1]: Is this file correct located? For the most tasks we located the tests under `w05h03/test/pgdp` (if you change the file location, you might need to update the first line in the test file)


image

**Comment by HendrikHuebner** [comment]
LGTM. I rewrote these tests using reflection. Im going to create a PR with my version once this has been merged if thats ok.

File: w05h03/MessengerTests.java (hunk 1 above)
Comment (nilsreichardt) [reply w05h03/MessengerTests.java]: Thanks for updating!

**Comment by MaximilianBoehmichen** [comment]
> LGTM. I rewrote these tests using reflection. Im going to create a PR with my version once this has been merged if thats ok.

Well, my tests work (apparently), so are there significant advantages of your version?

**Comment by HendrikHuebner** [comment]
Structural tests and not throwing compiler errors when one of the methods isn't implemented
